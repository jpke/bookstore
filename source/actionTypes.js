export const SELECT_NEW_BOOK = 'SELECT_NEW_BOOK';
export const ADMIN_ADD_BOOK = 'ADMIN_ADD_BOOK';
export const USER_INFO = 'USER_INFO';
export const DELIVERY_AND_BILLING = 'DELIVERY_AND_BILLING';
export const DELIVERY = 'DELIVERY';
export const BILLING = 'BILLING';
export const PAYMENT_INFO = 'PAYMENT_INFO';
export const SUBMIT_ORDER = 'SUBMIT_ORDER';
