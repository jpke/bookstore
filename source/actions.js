import * as types from './actionTypes';

export function selectNewBook(newBook) {
  return {
    type: types.SELECT_NEW_BOOK,
    newBook
  };
}

function deliveryAddress(info, useDeliveryAddressAsBilling) {
  return {
    type: types.DELIVERY,
    info,
    useDeliveryAddressAsBilling
  };
}

function billingAddress(info) {
  return {
    type: types.BILLING,
    info,
  };
}

export function submitInfo(info, type, option) {
  if(type === 'AdminAddBook') {
    if(info.title === "" || info.author === "") return;
    return {
      type: types.ADMIN_ADD_BOOK,
      info
    };
  }
  else if(type === 'UserInfo') {
    return {
      type: types.USER_INFO,
      info
    };
  }
  else if(type === 'Delivery') {
    if(option) {
      //use delivery address as billing address
      return {
        type: types.DELIVERY_AND_BILLING,
        info,
        useDeliveryAddressAsBilling: option
      };
    }
    else {
      return {
        type: types.DELIVERY,
        info,
        useDeliveryAddressAsBilling: option
      };
    }
  }
  else if(type === 'Billing') {
    return {
      type: types.BILLING,
      info,
    };
  }
  else if(type === 'PaymentInfo') {
    return {
      type: types.PAYMENT_INFO,
      info
    };
  }
  else if(type === 'SubmitOrder') {
    return {
      type: types.SUBMIT_ORDER,
      info
    };
  }

}
