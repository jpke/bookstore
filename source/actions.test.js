import * as types from './actionTypes';
import * as actions from './actions';

it('should create an action to select a book', () => {
  expect(actions.selectNewBook(
    {
      title: "The Evolution of Civilizations",
      author: "Carroll Quigley"
    }
  )).toMatchSnapshot();
})

it('should create an action to add a new available book', () => {
  expect(actions.submitInfo(
    {
      title: "New Book",
      author: "Famous Author"
    }, 'AdminAddBook'
  )).toMatchSnapshot();
})

it('should create an action to update state with user info', () => {
  expect(actions.submitInfo(
    {
      firstName: "_",
      lastName: "_",
      email: "_"
    }, 'UserInfo'
  )).toMatchSnapshot();
})

it('should create an action to update state with identical delivery and billing addresses', () => {
  expect(actions.submitInfo(
    {
      lineOne: "_",
      lineTwo: "_",
      city: "_",
      state: "_",
      zip: "_"
    }, 'Delivery', true
  )).toMatchSnapshot();
})

it('should create an action to update state with delivery address', () => {
  expect(actions.submitInfo(
    {
      lineOne: "_",
      lineTwo: "_",
      city: "_",
      state: "_",
      zip: "_"
    }, 'Delivery', false
  )).toMatchSnapshot();
})

it('should create an action to update state with billing address', () => {
  expect(actions.submitInfo(
    {
      lineOne: "_",
      lineTwo: "_",
      city: "_",
      state: "_",
      zip: "_"
    }, 'Billing'
  )).toMatchSnapshot();
})

it('should create an action to update state with payment info', () => {
  expect(actions.submitInfo(
    {
      nameOnCard: "_",
      cardNumber: "_",
      securityCode: "_"
    }, 'PaymentInfo'
  )).toMatchSnapshot();
})

it('should create an action to update state with submitted order', () => {
  expect(actions.submitInfo(
    {
      title: "The Evolution of Civilizations",
      author: "Carroll Quigley",
      date: new Date("Sat Mar 18 2017 16:38:08 GMT-0400 (EDT)")
    }, 'SubmitOrder'
  )).toMatchSnapshot();
})
