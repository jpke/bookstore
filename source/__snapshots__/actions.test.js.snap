// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create an action to add a new available book 1`] = `
Object {
  "info": Object {
    "author": "Famous Author",
    "title": "New Book",
  },
  "type": "ADMIN_ADD_BOOK",
}
`;

exports[`should create an action to select a book 1`] = `
Object {
  "newBook": Object {
    "author": "Carroll Quigley",
    "title": "The Evolution of Civilizations",
  },
  "type": "SELECT_NEW_BOOK",
}
`;

exports[`should create an action to update state with billing address 1`] = `
Object {
  "info": Object {
    "city": "_",
    "lineOne": "_",
    "lineTwo": "_",
    "state": "_",
    "zip": "_",
  },
  "type": "BILLING",
}
`;

exports[`should create an action to update state with delivery address 1`] = `
Object {
  "info": Object {
    "city": "_",
    "lineOne": "_",
    "lineTwo": "_",
    "state": "_",
    "zip": "_",
  },
  "type": "DELIVERY",
  "useDeliveryAddressAsBilling": false,
}
`;

exports[`should create an action to update state with identical delivery and billing addresses 1`] = `
Object {
  "info": Object {
    "city": "_",
    "lineOne": "_",
    "lineTwo": "_",
    "state": "_",
    "zip": "_",
  },
  "type": "DELIVERY_AND_BILLING",
  "useDeliveryAddressAsBilling": true,
}
`;

exports[`should create an action to update state with payment info 1`] = `
Object {
  "info": Object {
    "cardNumber": "_",
    "nameOnCard": "_",
    "securityCode": "_",
  },
  "type": "PAYMENT_INFO",
}
`;

exports[`should create an action to update state with submitted order 1`] = `
Object {
  "info": Object {
    "author": "Carroll Quigley",
    "date": 2017-03-18T20:38:08.000Z,
    "title": "The Evolution of Civilizations",
  },
  "type": "SUBMIT_ORDER",
}
`;

exports[`should create an action to update state with user info 1`] = `
Object {
  "info": Object {
    "email": "_",
    "firstName": "_",
    "lastName": "_",
  },
  "type": "USER_INFO",
}
`;
